src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/CustomConverterTests/CustomConverterTests.cs

```diff
@@ -179,40 +179,6 @@ void RunTest<TConverterReturn>()
             }
         }
 
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/66232", TargetFrameworkMonikers.NetFramework)]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/66371", typeof(PlatformDetection), nameof(PlatformDetection.IsMonoInterpreter))] 
-        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
-        public static void GetConverter_Poco_WriteThrowsNotSupportedException()
```

Comment:
This is a regression test validating .NET 6 behaviour. Calling into RUC methods is not equivalent to opting into reflection metadata resolution (though the two concerns often overlap). See also my comments in https://github.com/dotnet/runtime/pull/68496/files#r859817104 and https://github.com/dotnet/runtime/pull/68496/files#r859833346.

As mentioned, it might make sense to change this behavior, but we do need to file a breaking change document and provide knobs for getting the old behavior back in sourcegen. But it's certainly not in scope for a refactoring PR.

Reply:
We don't want a call to `JsonSerializerOptions.GetConverter` to trigger reflection-based metadata resolution for the instance. Most importantly though, it should not cause every other options instance in the process to opt in to reflection-based metadata resolution. That's what the final assertion is meant to check.